include/openthread/ip6.h

```diff
@@ -149,19 +139,13 @@ typedef struct otSockAddr
  */
 typedef struct otMessageInfo
 {
-    otIp6Address mSockAddr;    ///< The local IPv6 address.
-    otIp6Address mPeerAddr;    ///< The peer IPv6 address.
-    uint16_t     mSockPort;    ///< The local transport-layer port.
-    uint16_t     mPeerPort;    ///< The peer transport-layer port.
-    int8_t       mInterfaceId; ///< The IPv6 interface identifier.
-    uint8_t      mHopLimit;    ///< The IPv6 hop limit.
-
-    /**
-     * A pointer to link-specific information. In case @p mInterfaceId is set to OT_NETIF_INTERFACE_ID_THREAD,
-     * @p mLinkInfo points to otThreadLinkInfo(). This field is only valid for messages received from the
-     * Thread radio and is ignored on transmission.
-     */
-    const void *mLinkInfo;
+    otIp6Address mSockAddr;            ///< The local IPv6 address.
+    otIp6Address mPeerAddr;            ///< The peer IPv6 address.
+    uint16_t     mSockPort;            ///< The local transport-layer port.
+    uint16_t     mPeerPort;            ///< The peer transport-layer port.
+    const void * mLinkInfo;            ///< A pointer to link-specific information.
+    uint8_t      mHopLimit;            ///< The IPv6 hop limit.
+    bool         mIsHostInterface : 1; ///< TRUE if packets sent/received via host interface, FALSE otherwise.
```

Comment:
`PLATFORM_UDP` and `UDP_FORWARD` should be exclusive. Could we add a check in `openthread-core-config-check.h` to verify they are not both set? So that we can use this flag when PLATFORM_UDP is set.